test/e2e/upgrade/upgrade.go

```diff
@@ -422,10 +458,16 @@ func clusterUpgrade(f *framework.Framework, c configv1client.Interface, dc dynam
 				}
 				allUpdated := true
 				for _, p := range pools.Items {
-					updated, err := IsPoolUpdated(mcps, p.GetName())
-					if err != nil {
-						framework.Logf("error checking pool %s: %v", p.GetName(), err)
-						return false, nil
+					updated := false
+					// Don't check for worker pool update if its paused
+					if upgradeWorkerPoolPaused && p.GetName() == "worker" {
```

Comment:
Hrm.  I would expect worker pools to be paused as part of the preconfiguration for the job.  This feels more like part of the step registry configuration for the multi-step configuration.  Then this code would only need to skip paused pools.

Reply:
Right, lets move this part to step-registry instead

Reply:
Reworked, PTAL